### PR TITLE
add --etcd-cache-entries for user control of api server etcd decode cache size

### PR DESCRIPTION
**Pull request by sjenning**
@ncdc

### CODE REVIEW

**Comment by ncdc** [comment]
@kubernetes/rh-cluster-infra @smarterclayton @liggitt @kubernetes/sig-api-machinery

**Comment by hongchaodeng** [comment]
cc @xiang90

File: pkg/storage/etcd/etcd_helper.go [hunk 1]
```diff
@@ -154,7 +155,6 @@ type etcdHelper struct {
 	// This depends on etcd's indexes being globally unique across all objects/types. This will
```
Comment (liggitt) [inline pkg/storage/etcd/etcd_helper.go]: this comment is concerning... we already allow segmenting by type, which means we can't assume an etcd index is globally unique

File: pkg/master/master_test.go [hunk 2]
```diff
@@ -68,13 +68,13 @@ func setUp(t *testing.T) (*Master, *etcdtesting.EtcdTestServer, Config, *assert.
...
+		api.GroupName, etcdstorage.NewEtcdStorage(server.Client, testapi.Default.Codec(), etcdtest.PathPrefix(), false, etcdstorage.DefaultCacheEntires))
 	storageDestinations.AddAPIGroup(
-		autoscaling.GroupName, etcdstorage.NewEtcdStorage(server.Client, testapi.Autoscaling.Codec(), etcdtest.PathPrefix(), false))
+		autoscaling.GroupName, etcdstorage.NewEtcdStorage(server.Client, testapi.Autoscaling.Codec(), etcdtest.PathPrefix(), false, etcdstorage.DefaultCacheEntires))
 	storageDestinations.AddAPIGroup(
-		batch.GroupName, etcdstorage.NewEtcdStorage(server.Client, testapi.Batch.Codec(), etcdtest.PathPrefix(), false))
+		batch.GroupName, etcdstorage.NewEtcdStorage(server.Client, testapi.Batch.Codec(), etcdtest.PathPrefix(), false, etcdstorage.DefaultCacheEntires))
 	storageDestinations.AddAPIGroup(
-		extensions.GroupName, etcdstorage.NewEtcdStorage(server.Client, testapi.Extensions.Codec(), etcdtest.PathPrefix(), false))
+		extensions.GroupName, etcdstorage.NewEtcdStorage(server.Client, testapi.Extensions.Codec(), etcdtest.PathPrefix(), false, etcdstorage.DefaultCacheEntires))
```
Comment (liggitt) [inline pkg/master/master_test.go]: this appears to set max cache entries per storage destination, so max entries is effectively 4*max

Comment (sjenning) [inline pkg/master/master_test.go]: yes, that is true

File: pkg/storage/etcd/etcd_helper.go (hunk 1 above)
Comment (sjenning) [inline pkg/storage/etcd/etcd_helper.go]: that is a problem then.  a preexisting one that is; not one that this PR creates.

**Comment by lavalamp** [comment]
I'm confused, I think there's another PR doing this exact thing in the queue right now?

**Comment by lavalamp** [comment]
Yes, #23914 is already in progress :)

**Comment by ncdc** [comment]
@lavalamp sorry about that! We were working an OpenShift bug for the same thing and didn't know there was already something in flight.

**Pull request closed without merging**